concent_api/core/tests/test_integration_accept_or_reject_results.py

```diff
@@ -0,0 +1,428 @@
+from django.urls            import reverse
+from freezegun              import freeze_time
+from golem_messages         import message
+
+from core.models            import Message
+from core.models            import ReceiveStatus
+from core.tests.utils       import ConcentIntegrationTestCase
+from utils.testing_helpers  import generate_ecc_key_pair
+
+
+(CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+def get_provier_account_status_true_mock(_):
+    return True
+
+
+def get_provier_account_status_false_mock(_):
+    return False
+
+
+@override_settings(
+    CONCENT_PRIVATE_KEY    = CONCENT_PRIVATE_KEY,
+    CONCENT_PUBLIC_KEY     = CONCENT_PUBLIC_KEY,
+    CONCENT_MESSAGING_TIME = 10,  # seconds
+    FORCE_ACCEPTANCE_TIME  = 10,  # seconds
+)
+class GetTaskResultIntegrationTest(ConcentIntegrationTestCase):
+
+    def test_provider_forces_subtask_results_for_task_which_was_already_submitted_concent_should_refuse(self):
+        """
+        Tests if on provider ForceSubtaskResults message Concent will return ServiceRefused
+        if ForceSubtaskResults with same task_id was already submitted.
+
+        Expected message exchange:
+        Provider  -> Concent:    ForceSubtaskResults
+        Concent   -> Provider:   HTTP 202
+        Provider  -> Concent:    ForceSubtaskResults
+        Concent   -> Provider:   ServiceRefused
+        """
+
+        # STEP 1: Provider forces subtask results via Concent.
+        # Request is processed correctly.
+        serialized_force_subtask_results = self._get_serialized_force_subtask_results(
+            timestamp                   = "2018-02-05 10:00:15",
+            ack_report_computed_task    = self._get_deserialized_ack_report_computed_task(
+                subtask_id      = "xxyyzz",
+                task_to_compute = self._get_deserialized_task_to_compute(
+                    timestamp   = "2018-02-05 10:00:00",
+                    deadline    = "2018-02-05 10:00:10",
+                    task_id     = 2,
+                )
+            )
+        )
+
+        with mock.patch('core.views.get_provider_account_status', get_provier_account_status_true_mock):
+            with freeze_time("2018-02-05 10:00:30"):
+                response_1 = self.client.post(
+                    reverse('core:send'),
+                    data                                = serialized_force_subtask_results,
+                    content_type                        = 'application/octet-stream',
+                    HTTP_CONCENT_CLIENT_PUBLIC_KEY      = self._get_encoded_provider_public_key(),
+                )
+
+        self.assertEqual(Message.objects.last().type,               message.concents.ForceSubtaskResults.TYPE)
+        self.assertEqual(Message.objects.last().task_id,            2)
+        self.assertEqual(ReceiveStatus.objects.last().delivered,    False)
+        self.assertEqual(response_1.status_code,                    202)
```

Comment:
Why not using test_response here too ? This is possible for current implementation.